cookbooks/fb_consul/recipes/default.rb

```diff
@@ -0,0 +1,157 @@
+  'shell' => '/usr/sbin/nologin',
+}
+
+package 'consul' do
+  action :upgrade
+  notifies :restart, 'service[consul]'
+end
+
+whyrun_safe_ruby_block 'validate config' do
+  block do
+    node['fb_consul']['config'].each_key do |key|
+      if ['config-file', 'config-dir'].include?(key)
+        fail "fb_consul::default: #{key} is not allowed in 'config'! " +
+          'Please use "services" and "checks" to configure consul.'
+      end
+    end
+  end
+end
+
+directory 'consul data dir' do
+  only_if do
+    node['fb_consul']['enabled']
+  end
+  path lazy { node['fb_consul']['config']['data_dir'] }
+  owner 'consul'
+  group 'root'
+  mode '0770'
+end
+
+execute 'fix_perms' do
+  command 'chown -R consul /var/lib/consul'
+  action :nothing
+end
+
+file '/etc/systemd/system/consul.service' do
+  action :delete
+  notifies :reload, 'fb_systemd_reload[system instance]', :immediately
+  notifies :run, 'execute[fix_perms]', :immediately
+end
+
+directory '/etc/consul' do
+  owner 'root'
+  group 'root'
+  mode '0755'
+end
+
+cookbook_file '/etc/default/consul' do
+  source 'consul.default'
+  owner 'root'
+  group 'root'
+  mode '0644'
+end
+
+cookbook_file '/etc/consul/consul-agent-ca.pem' do
+  only_if { node['fb_consul']['certificate_cookbook'] }
+  cookbook lazy { node['fb_consul']['certificate_cookbook'] }
+  owner 'root'
+  group 'root'
+  mode '0644'
+end
+
+cookbook_file '/etc/consul/consul-agent-ca-key.pem' do
+  only_if do
+    node['fb_consul']['config']['server'] &&
+    node['fb_consul']['certificate_cookbook']
+  end
+  cookbook lazy { node['fb_consul']['certificate_cookbook'] }
+  owner 'consul'
+  group 'root'
+  mode '0600'
+end
+
+cookbook_file '/etc/consul/consul-server.pem' do
+  only_if do
+    node['fb_consul']['config']['server'] &&
+    node['fb_consul']['certificate_cookbook']
+  end
+  cookbook lazy { node['fb_consul']['certificate_cookbook'] }
+  source "consul-server-#{node['hostname']}.pem"
+  owner 'root'
+  group 'root'
+  mode '0644'
+end
+
+cookbook_file '/etc/consul/consul-server-key.pem' do
+  only_if do
+    node['fb_consul']['config']['server'] &&
+    node['fb_consul']['certificate_cookbook']
+  end
+  cookbook lazy { node['fb_consul']['certificate_cookbook'] }
+  source "consul-server-key-#{node['hostname']}.pem"
+  owner 'consul'
+  group 'root'
+  mode '0600'
+end
+
+whyrun_safe_ruby_block 'add crypto options' do
+  only_if { node['fb_consul']['certificate_cookbook'] }
+  block do
+    node.default['fb_consul']['config']['ca_file'] =
+      '/etc/consul/consul-agent-ca.pem'
+    if node['fb_consul']['config']['server']
+      node.default['fb_consul']['config']['cert_file'] =
+        '/etc/consul/consul-server.pem'
+      node.default['fb_consul']['config']['key_file'] =
+        '/etc/consul/consul-server-key.pem'
+    end
+  end
+end
+
+template '/etc/consul/consul.json' do
+  owner 'root'
+  group 'root'
+  mode '0644'
+  source 'consul.json.erb'
+  verify '/usr/bin/consul validate %{path}'
```

Comment:
I don't think this'll work, `path` is undefined here, and moreover you're using a non-interpolated string

Reply:
.... you haven't had your coffee yet man. 

```
[phil@rider (chef13) cookbooks]$ grep -r 'verify' * | grep path
fb_dnsmasq/recipes/default.rb:  verify 'dnsmasq --test -C %{path}'
fb_iptables/recipes/default.rb:  verify do |path|
fb_iptables/recipes/default.rb:  verify do |path|
fb_sudo/recipes/default.rb:  verify 'visudo -c -q -f %{path}'
```

Reply:
TIL